lib/features/home_page/widgets/pokemon_tile.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:cached_network_image/cached_network_image.dart';
+import 'package:flutter/material.dart';
+import 'package:pokedex/api/api_service.dart';
+import 'package:pokedex/api/models/pokemon_model.dart';
+import 'package:pokedex/api/models/pokemon_type_model.dart';
+import 'package:pokedex/features/details_page/details_page_connector.dart';
+import 'package:pokedex/features/home_page/widgets/type.dart';
+import 'package:pokedex/utilities/app_starter.dart';
+import 'package:pokedex/utilities/constants.dart';
+import 'package:pokedex/utilities/extensions.dart';
+
+class PokemonTile extends StatefulWidget {
+  const PokemonTile({
+    this.pokemon,
+    Key? key,
+  }) : super(key: key);
+
+  final PokemonModel? pokemon;
+
+  @override
+  State<PokemonTile> createState() => _PokemonTileState();
+}
+
+class _PokemonTileState extends State<PokemonTile> {
+  List<PokemonTypeModel>? types;
+  String? url;
+
+  void _getTypes() async {
+    await getIt<ApiService>().pokemonApi.getPokemonType(widget.pokemon?.url?.pokemonId ?? 0).then((value) {
+      if (!mounted) return;
+      setState(() => types = value);
+    });
+  }
+
+  @override
+  void initState() {
+    _getTypes();
+
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final textTheme = Theme.of(context).textTheme;
+
+    return Container(
+      decoration: BoxDecoration(
+        color: types?.first.name.toString().pokemonColor,
+        borderRadius: BorderRadius.circular(25),
+        boxShadow: [
+          BoxShadow(
+            color: Colors.black26.withOpacity(0.2),
+            offset: const Offset(0.0, 0.0),
+            spreadRadius: 0.10,
+          )
+        ],
+      ),
+      child: (types != null)
+          ? Material(
```

Comment:
Could you check if you could remove `Material`

Reply:
i use it for the ripple effect when a tile is tapped

Reply:
https://user-images.githubusercontent.com/61373410/184803604-0ba19292-39d5-496d-aaee-6147cb12c4e3.mp4